### PR TITLE
bundler(compile): Patch for infinite download loops

### LINKED ISSUE #278: Infinite download loop on start / build
When importing a module within which certain files indirectly import themselves, Deno can handle the import fine; however, `aleph start` or `aleph build` goes into an infinite download loop of these files.

This is the import that is breaking Aleph for me:
```typescript
import { api } from "https://gitlab.com/silk-matrix/matrix.ts/-/raw/0.1.6/mod.ts";
```

Edit 1: I did some searching through the Aleph code, this is the line of the log that indicates that there is an infinite download loop:
https://github.com/alephjs/aleph.js/blob/v0.3.0-alpha.29/server/app.ts#L924

Edit 2: After some more digging I found where the function is that infinitely calls itself at this point:
https://github.com/alephjs/aleph.js/blob/v0.3.0-alpha.29/bundler/mod.ts#L171

### PR DESCRIPTION
**Pull request by Gleb-ko**
When there is an external module that imports from another external module which imports from the first, the bundler goes into an infinite loop of compiling the two modules. This change fixes this issue and as far as I can see this change has no unintended side-effects.

Closes #278

### CODE REVIEW

**Comment by shadowtime2000** [comment]
@Gleb-ko Do you think you could add a test that does a reproduction of the bug to test if this fix works? This also helps to m ake sure the bug doesn't come back in future releases

**Comment by Gleb-ko** [comment]
> @Gleb-ko Do you think you could add a test that does a reproduction of the bug to test if this fix works? This also helps to m ake sure the bug doesn't come back in future releases

@shadowtime2000 Do you mean a manual test in the examples/ or an automated test?

**Comment by shadowtime2000** [comment]
@Gleb-ko I mean an automated test

**Comment by Gleb-ko** [comment]
@shadowtime2000 oh okay, sorry in that case I don't see a non-hacky way to test fur an infinite recursion without changing the actual source code.

**Comment by shadowtime2000** [comment]
@Gleb-ko What do you mean by the "actual source code"

**Comment by Gleb-ko** [comment]
@shadowtime2000 I mean the code for the bundler itself, since the only way I see to test whether the compile function is calling itself too much is to have some sort of counter. Since this counter would only be used for a test it seems quite inefficient to do this to me. The only other way I see to do it is to just have a timeout, which is quite hacky, but I suppose it could be fine for just a test.

**Comment by shadowtime2000** [comment]
@Gleb-ko I think  a counter should be fine because it shouldn't slow down the compiler that much.

File: bundler/mod.ts [hunk 1]
```diff
@@ -167,7 +174,7 @@ export class Bundler {
         const depMod = this.#app.getModule(dep.url)
         if (depMod !== null) {
           const s = `.bundling.js#${dep.url}@`
-          await this.compile(depMod, external)
+          await this.compile(depMod, [dep.url, external].flat())
```
Comment (ije) [inline bundler/mod.ts:177]: in dev mode, we use a map to record the module urls before it's going to be compiled. that can avoid loop import issue. here better not change the external list.

**Comment by ije** [comment]
@Gleb-ko i'm going to close this since i already fixed it in **alpha.30** https://github.com/alephjs/aleph.js/commit/a24c4cdcba479a22a86f4ccc1b94ccc2b9177c2b, thanks for your great works that helps a lot.

**Pull request closed without merging**